internal/fs/inode/dir.go

```diff
@@ -1048,6 +1052,45 @@ func (d *dirInode) LocalFileEntries(localFileInodes map[Name]Inode) (localEntrie
 	return
 }
 
+// LOCKS_REQUIRED(fs)
+func (d *dirInode) LocalFileEntriesPlus(localFileInodes map[Name]Inode) (localEntries map[string]fuseutil.DirentPlus) {
+	localEntries = make(map[string]fuseutil.DirentPlus)
+
+	for localInodeName, in := range localFileInodes {
+		// It is possible that the local file inode has been unlinked, but
+		// still present in localFileInodes map because of open file handle.
+		// So, if the inode has been unlinked, skip the entry.
+		file, ok := in.(*FileInode)
+		if ok && file.IsUnlinked() {
+			continue
+		}
+		if localInodeName.IsDirectChildOf(d.Name()) {
+			in.Lock()
```

Comment:
You need to follow the lock ordering here. iNode.Lock() should be taken before fs.Lock(), seems we are not following here.